add hw04

Efefefef opened the pull request:
- follow Leon's tutorial
- fix errors
- wasn't able to run due to CUDA_ERROR_OUT_OF_MEMORY, please see if you can handle it or find the proper hyperparamerets

Efefefef committed:
add hw04

imtezcan commented on homework04/breakout.py:
```diff
@@ -0,0 +1,228 @@
+    returns = np.zeros(num_parallel_tests)
+
+    while not done:
+        q_values = test_dqn_network(states)
```
with this version I get the following error:
```
ValueError: Input 0 of layer "model" is incompatible with the layer: expected shape=(None, 84, 84, 3), found shape=(10, 210, 160, 3)
```

based on our discussion I made the following change and was able to run it, so let's change it here too:
```suggestion
        q_values = test_dqn_network(observation_preprocessing_function(states))
```

Commit:
fix - preprocess observations before test

to fix the error
```ValueError: Input 0 of layer "model" is incompatible
with the layer: expected shape=(None, 84, 84, 3), found shape=(10, 210,
160, 3)
```
